File: src/main/java/com/github/onlaait/mcserverpinger/AddressResolver.java
package com.github.onlaait.mcserverpinger;

import org.slf4j.Logger;

import java.net.InetAddress;
import java.net.InetSocketAddress;
import java.net.UnknownHostException;
import java.util.Optional;

@FunctionalInterface
public interface AddressResolver {
    Logger LOGGER = Log.INSTANCE.getLogger();
    AddressResolver DEFAULT = address -> {
        try {
            InetAddress inetAddress = InetAddress.getByName(address.getAddress());
            return Optional.of(Address.create(new InetSocketAddress(inetAddress, address.getPort())));
        } catch (UnknownHostException unknownHostException) {
            LOGGER.debug("Couldn't resolve server {} address", address.getAddress());
            return Optional.empty();
        }
    };

    Optional<Address> resolve(ServerAddress var1);
}

File: src/main/kotlin/mcserverpinger/address/AddressResolver.kt
package com.github.onlaait.mcserverpinger.address

import com.github.onlaait.mcserverpinger.Log.logger
import org.slf4j.Logger
import java.net.InetAddress
import java.net.InetSocketAddress
import java.net.UnknownHostException

object AddressResolver {

    val LOGGER: Logger = logger

    fun resolve(address: ServerAddress): List<InetSocketAddress>? {
        try {
            val inetAddresses = InetAddress.getAllByName(address.address)
            return inetAddresses.map { InetSocketAddress(it, address.port) }
        } catch (_: UnknownHostException) {
            LOGGER.debug("Couldn't resolve server ${address.address} address")
            return null
        }
    }
}